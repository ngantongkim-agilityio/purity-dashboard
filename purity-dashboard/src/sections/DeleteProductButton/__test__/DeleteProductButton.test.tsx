import { render } from '@testing-library/react';

// Components
import { DeleteProductButton } from '..';

jest.mock('@/services/products.ts', () => ({
  fetchLatestProducts: jest.fn()
}));

describe('DeleteProductButton component', () => {
  const mockProps = {
    id: '1'
  };

  it('Should render DeleteProductButton component with default props', () => {
    const { container } = render(<DeleteProductButton {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
