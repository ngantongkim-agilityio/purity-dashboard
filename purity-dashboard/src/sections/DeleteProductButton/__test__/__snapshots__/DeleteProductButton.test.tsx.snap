// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteProductButton component Should render DeleteProductButton component with default props 1`] = `
<div>
  <form
    action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
  >
    <button
      class="rounded-md border p-2 hover:bg-gray-100"
      type="submit"
    >
      <span
        class="sr-only"
      >
        Delete
      </span>
      <svg
        aria-hidden="true"
        class="w-5"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </form>
</div>
`;
